sdk/deviceupdate/iot-device-update-rest/test/public/utils/recordedClient.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createTestCredential } from "@azure-tools/test-credential";
+import DeviceUpdate, { DeviceUpdateRestClient } from "../../../src";
+import { Recorder, assertEnvironmentVariable } from "@azure-tools/test-recorder";
+
+export function createRecordedClient(recorder: Recorder): DeviceUpdateRestClient {
+  const credential = createTestCredential();
+  const client = DeviceUpdate(assertEnvironmentVariable("ENDPOINT"), credential);
+  recorder.configureClient(client);
```

Comment:
`recorder.configureClient(client);` call is likely going to change based on the discussion at https://github.com/Azure/azure-sdk-for-js/issues/16876 with the Arch board.

Basically, we don't want to expose "pipeline" object on the clients, which this client is currently doing.
We're instead going to go with an approach of adding policies in an array in the client options.
Change being made here - https://github.com/Azure/azure-sdk-for-js/pull/19920

Reply:
@xirzec this client exposes "pipeline", we don't want that, right?

Reply:
@HarshaNalluru I think we don't need to use adding polices in this PR as the #19920 is not merged. If it's ready to use policies and you need my help, please let me know. For now, I will merge this PR to release the sdk. thankx

Reply:
My comment was about the already exposed "pipeline" object on the client. 🙂
Many clients exposed it unintentionally, we're going to make a call for those - mostly making it private or something similar. 
And this #19920 is to provide an alternative to allow adding policies.

Since there are other clients that are in a similar state as iot-device-update, we'll handle this package along with them if we ever decide to change something even if it is a breaking change.